### PR TITLE
Add builders for creating interactions

### PR DESCRIPTION
**Pull request by Th3-M4jor**
While this is grossly incomplete, and still needs a lot more documentation,
I would like to get feedback on what's there before continuing.

### CODE REVIEW

**Comment by Th3-M4jor** [comment]
I believe this is ready for review.
Could add functions for creating and sending these to `model::interactions` but I think that should wait until after #1175 is merged

File: src/builder/create_interaction.rs
Comment (arqunis) [inline src/builder/create_interaction.rs]:
```suggestion
        self.add_sub_option(data)
```

File: src/builder/create_interaction.rs
Comment (arqunis) [inline src/builder/create_interaction.rs]:
```suggestion
        self.add_interaction_option(data)
```

**Review by arqunis — CHANGES_REQUESTED** [review]
This is good, but is there a reason the target branch is `next`? I don't see why this addition should have to be released in 0.11.

**Comment by Th3-M4jor** [comment]
You're right, this probably can be merged into `next` and `current`.

**Review by arqunis — APPROVED** [review]

**Comment by arqunis** [comment]
Manually rebased onto `current`: https://github.com/serenity-rs/serenity/commit/0d743fdd3b7cc0ea28500547ff0a9caff4a83c3f

**Pull request closed without merging**
